lib/UI.js

```diff
@@ -108,17 +109,40 @@ define(function (require, exports, module) {
             // If it is, the attached preview is hidden
             // and the detached preview is opened.
             if(Resizer.isVisible("#second-pane")) {
-                Resizer.hide("#second-pane");
-                $("#first-pane").addClass("expandEditor");
-                IframeBrowser.detachPreview();
+                _detachPreview();
             }
             else {
-                Resizer.show("#second-pane");
-                $("#first-pane").removeClass("expandEditor");
+                _attachPreview();
             }
         });
     }
 
+    /**
+      * Used to hide second pane, spawn detached preview, and attach beforeunload listener
+      */
+    function _detachPreview() {
+        Resizer.hide("#second-pane");
+        $("#first-pane").addClass("expandEditor");
+        $('#liveDevButton').removeClass("liveDevButtonDetach");
+        $('#liveDevButton').addClass("liveDevButtonAttach");
+        var detachedWindow = IframeBrowser.detachPreview();
+        detachedWindow.addEventListener("beforeunload", _attachPreview, false);
+    }
+
+    /**
+      * Used to show second pane, change lilveDevButton background and close the detached preview
+      */
+    function _attachPreview() {
+        Resizer.show("#second-pane");
+        $('#liveDevButton').removeClass("liveDevButtonAttach");
+        $('#liveDevButton').addClass("liveDevButtonDetach");
+        $("#first-pane").removeClass("expandEditor");
+        var detachedWindow = IframeBrowser.getDetachedPreview();
+        if(detachedWindow) {
+          detachedWindow.close();
```

Comment:
Alright did a little more testing, and yes, the function does run twice, working on a fix now